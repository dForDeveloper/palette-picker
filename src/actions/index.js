export const setColors = (colors) => ({
  type: 'SET_COLORS',
  colors
});

export const setLockedColors = (lockedColors) => ({
  type: 'SET_LOCKED_COLORS',
  lockedColors
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS',
  projects
});

export const toggleLoading = (bool) => ({
  type: 'TOGGLE_LOADING',
  bool
});

export const setError = (message) => ({
  type: 'SET_ERROR',
  message
});

export const setPalettes = (id, palettes) => ({
  type: 'SET_PALETTES',
  id,
  palettes
});

export const setModal = (isDisplayed, modalType = null, text = null) => ({
  type: 'SET_MODAL',
  isDisplayed,
  modalType,
  text
});