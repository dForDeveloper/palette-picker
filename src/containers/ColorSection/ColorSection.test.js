import React from 'react';
import { shallow } from 'enzyme';
import { setLockedColors } from '../../actions';
import {
  ColorSection,
  mapStateToProps,
  mapDispatchToProps
} from './ColorSection';

describe('ColorSection', () => {
  let wrapper;
  const mockProps = {
    colors: ['#aabbcc', '#bbccdd', '#ccddee', '#778899', '#ffffff'],
    lockedColors: [0],
    color: '#ffffff',
    index: 0,
    setLockedColors: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<ColorSection {...mockProps} />);
  });

  describe('render', () => {
    it('should match the snapshot when the color is light/locked', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the color is dark/locked', () => {
      wrapper = shallow(<ColorSection {...mockProps} color='#000000' />);
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should match the snapshot when the color is light/unlocked', () => {
      wrapper = shallow(<ColorSection {...mockProps} lockedColors={[]} />);      
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the color is dark/unlocked', () => {
      wrapper = shallow(
        <ColorSection {...mockProps} color='#000000' lockedColors={[]} />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleClick when the section is clicked', () => {
      jest.spyOn(wrapper.instance(), 'handleClick');
      wrapper.instance().forceUpdate();
      wrapper.find('section').simulate('click', { target: { id: '#ffffff' } });
      expect(wrapper.instance().handleClick).toHaveBeenCalled();
    });
  });

  describe('handleClick', () => {
    it('should call setLockedColors when unlocked', () => {
      wrapper.instance().handleClick({ target: { id: '#ffffff' } });
      expect(mockProps.setLockedColors).toHaveBeenCalledWith([0 , 4]);
    });
    
    it('should call setLockedColors when locked', () => {
      wrapper.instance().handleClick({ target: { id: '#aabbcc' } });
      expect(mockProps.setLockedColors).toHaveBeenCalledWith([]);
    });
  });
});

describe('mapStateToProps', () => {
  it('should return a props object with the correct properties', () => {
    const mockState = {
      colors: ['#333333'],
      lockedColors: [0],
      extraProperty: true
    };
    const expected = {
      colors: ['#333333'],
      lockedColors: [0]
    }
    const result = mapStateToProps(mockState);
    expect(result).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should dispatch setLockedColors', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    const expected = setLockedColors([0]);
    mappedProps.setLockedColors([0]);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});