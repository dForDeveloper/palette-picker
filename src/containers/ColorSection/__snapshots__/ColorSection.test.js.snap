// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSection render should match the snapshot when the color is dark/locked 1`] = `
<section
  className="section--locked-true-true"
  id="#000000"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#000000",
    }
  }
>
  <h3
    className="h3--dark-true"
  >
    #000000
  </h3>
</section>
`;

exports[`ColorSection render should match the snapshot when the color is dark/unlocked 1`] = `
<section
  className="section--locked-false-true"
  id="#000000"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#000000",
    }
  }
>
  <h3
    className="h3--dark-true"
  >
    #000000
  </h3>
</section>
`;

exports[`ColorSection render should match the snapshot when the color is light/locked 1`] = `
<section
  className="section--locked-true-false"
  id="#ffffff"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <h3
    className="h3--dark-false"
  >
    #ffffff
  </h3>
</section>
`;

exports[`ColorSection render should match the snapshot when the color is light/unlocked 1`] = `
<section
  className="section--locked-false-false"
  id="#ffffff"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <h3
    className="h3--dark-false"
  >
    #ffffff
  </h3>
</section>
`;
